### DISCUSSION TITLE
Disable the `curly` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#219] [General]
There is a proposal to disable the `curly` rule.

> `curly`: this rule is too strict, and one of the most annoying. One liners can keep the code cleaner and easier to read.
> \- @edgardz

**Comment by leroykorterink** [comment]
Allows the omission of curly braces when a block contains only one statement. However, it is considered by many to be best practice to never omit curly braces around blocks, even when they are optional, because it can lead to bugs and reduces code clarity. So the following:

```ts
if (foo) foo++;
```

Can be rewritten as:

```ts
if (foo) {
    foo++;
}
```

Some problems reported by this rule are automatically fixable. For more details see [the documentation](https://eslint.org/docs/latest/rules/curly)

**Comment by ThaNarie** [comment]
Let me get ahead that most comments against this aren't really relevant, since prettier already normalises a lot.

A common mistake is code like this:
```ts
if (foo)
  console.log('a');
  console.log('b');

somethingElse();
```

will only wrap the first log in the if, while the other is part of the normal code flow.

However, prettier already rewrites it as:
```ts
if (foo) console.log('a');
console.log('b');

somethingElse();
```

Or the people that say adding/removing braces when they want to add/remove a line: There are IDE quick-fixes for that (same as for arrow function braces).

---

I also do agree that it's preferred for small checks/statements like this:

```ts
if (conditionIsTrue) return;

if (isEnabled) callSomething()
```

However, things get a bit more verbose/condensed when looking at:

```ts
// nobody would write this, but it's allowed
if (a === true) if (b === true) alert(a);

// how do you parse the above clearly?
if (someFunc(passParam) > SOME_LIMIT) somethingElse(withAParam, 'a-setting');
```

In the last example, I find it difficult to parse when the condition ends, and the "then" part starts.

And I'm afraid that a lot of code will start looking like that :( Especially when having `else` in the mix:

```ts
if (someFunc(passParam) > SOME_LIMIT) somethingElse(withAParam, 'a-setting');
else if (someFunc(passParam, param2) > OTHER_LIMIT) somethingElse(withAParam);
else if (!otherFunc(passParam)) somethingElse(withAParam, 'a-setting');
else doCompletelyNothing('thequicbrownfox');
```

---

> One liners can keep the code cleaner and **easier to read**.

The issue (in all of these conversations) is that this differs from person to person. Where one lines to scan horizontally, others like to scan vertically. Where some like whitespace to clearly see separation in pieces of code, others like things more condensed to reduce scrolling.
